### PR TITLE
Feature: Ground state projection tool for RT-TDDFT

### PR DESCRIPTION
**Pull request by ESROAMER**
### Background
For Rt-TDDFT calculations, the ground-state projection is an important feature. By projecting the excited-state wavefunction at time t onto the reference state at time 0, real-time excitation infos can be obtained. Below, we use N2 as an example to demonstrate this feature.
By applying a Gaussian pulse around 2 fs, the electrons in N2 are excited and begin to oscillate. This oscillation can be observed by monitoring the electric dipole.
image
Meanwhile, the number of electrons in the ground state begins to decrease. By summing the number of ground-state electron
image
By analyzing the real-time projection matrix, we can clearly observe three main excitations:
HOMO-1 -> LUMO+1  
HOMO-2 -> LUMO  
HOMO -> LUMO+2  
result
These three excitations can be clearly seen in the absorption spectrum obtained via Fourier transformation.
image

### What's changed?
1. Created new folder 'rt-tddft-tools' The previous absorption tools have been moved there.
2. Uploaded the ground-state projection tool.
3. Uploaded the tool for calculating overlap matrices. In cases where atoms are moved, this script must be used to regenerate the overlap matrix before performing the ground-state projection correctly.
4. Uploaded an example for Si to demonstrate the use of the ground-state projection tool.

### Linked Issue
#4925

### CODE REVIEW

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/Excited_band_for_ik1.png
Comment (mohanchen) [inline tools/rt-tddft-tools/examples/ground-state-projection-Si/Excited_band_for_ik1.png:1]: why do we need this file, and the file is huge.

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/On_0.dat
Comment (mohanchen) [inline tools/rt-tddft-tools/examples/ground-state-projection-Si/On_0.dat:1]: the file is big, is there any way to reduce the size of the file?

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/On_1.dat
Comment (mohanchen) [inline tools/rt-tddft-tools/examples/ground-state-projection-Si/On_1.dat:1]: the file is large, too

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/Si_ONCV_PBE-1.0.upf
Comment (mohanchen) [inline tools/rt-tddft-tools/examples/ground-state-projection-Si/Si_ONCV_PBE-1.0.upf:1]: the file is too large, I suggest using the pseudopotential file that is already in the code

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/Si_gga_8au_100Ry_2s2p1d.orb
Comment (mohanchen) [inline tools/rt-tddft-tools/examples/ground-state-projection-Si/Si_gga_8au_100Ry_2s2p1d.orb:1]: file too large

File: tools/rt-tddft-tools/overlap_generator.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        orb_file_list = list(atoms0.info['basis'].values())
+        orb_file_list = [orb_file_dir + orbfile for orbfile in orb_file_list]
+        orb_file_num = len(orb_file_list)
+
+        symbols = atoms0.get_chemical_symbols()
+        element_counts = Counter(symbols)
+        unique_elements = set(symbols)
+        assert orb_file_num == len(unique_elements), f"轨道数目{orb_file_num}与元素数目{len(unique_elements)}不一致！"
+
+        # 构建轨道合集
```
Comment (mohanchen) [inline tools/rt-tddft-tools/overlap_generator.py]: do not use Chinese

File: tools/rt-tddft-tools/overlap_generator.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    def __init__(self, orb_file_dir, atoms0, atomst):
+        # 从结构中获取轨道信息
+        orb_file_list = list(atoms0.info['basis'].values())
+        orb_file_list = [orb_file_dir + orbfile for orbfile in orb_file_list]
+        orb_file_num = len(orb_file_list)
+
+        symbols = atoms0.get_chemical_symbols()
+        element_counts = Counter(symbols)
+        unique_elements = set(symbols)
+        assert orb_file_num == len(unique_elements), f"轨道数目{orb_file_num}与元素数目{len(unique_elements)}不一致！"
```
Comment (mohanchen) [inline tools/rt-tddft-tools/overlap_generator.py]: do not use Chinese

File: tools/rt-tddft-tools/overlap_generator.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+from pyabacus import ModuleBase as base
+import numpy as np
+from collections import Counter
+import pyatb
+from pyatb.easy_use import stru_analyzer
+import re
+Bohr=0.5291772105638411
+class overlap_R:
+    def __init__(self, orb_file_dir, atoms0, atomst):
+        # 从结构中获取轨道信息
```
Comment (mohanchen) [inline tools/rt-tddft-tools/overlap_generator.py]: do not use Chinese

File: tools/rt-tddft-tools/overlap_generator.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        symbols = atoms0.get_chemical_symbols()
+        element_counts = Counter(symbols)
+        unique_elements = set(symbols)
+        assert orb_file_num == len(unique_elements), f"轨道数目{orb_file_num}与元素数目{len(unique_elements)}不一致！"
+
+        # 构建轨道合集
+        self.orb = nao.RadialCollection()
+        self.orb.build(orb_file_num, orb_file_list, 'o')
+
+        # 统一轨道的格点
```
Comment (mohanchen) [inline tools/rt-tddft-tools/overlap_generator.py]: do not use Chinese

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/Si_ONCV_PBE-1.0.upf
Comment (ESROAMER) [reply tools/rt-tddft-tools/examples/ground-state-projection-Si/Si_ONCV_PBE-1.0.upf:1]: Okay, I'll delete the pseudopotential file and the orbital file.

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/On_0.dat
Comment (ESROAMER) [reply tools/rt-tddft-tools/examples/ground-state-projection-Si/On_0.dat:1]: This one will be deleted.

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/On_1.dat
Comment (ESROAMER) [reply tools/rt-tddft-tools/examples/ground-state-projection-Si/On_1.dat:1]: This one will keep only 10 bands and 100 steps of data.

File: tools/rt-tddft-tools/examples/ground-state-projection-Si/Excited_band_for_ik1.png
Comment (ESROAMER) [reply tools/rt-tddft-tools/examples/ground-state-projection-Si/Excited_band_for_ik1.png:1]: This image is used to display the results of the example. I can convert it to JPEG format, or maybe I should just delete it?

File: tools/rt-tddft-tools/overlap_generator.py (hunk 1 above)
Comment (ESROAMER) [reply tools/rt-tddft-tools/overlap_generator.py]: Okay.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**